File: frontend/src/types/property.ts
export type PropertyType = 'appartement' | 'maison' | 'local_commercial' | 'autre'

export const PROPERTY_TYPE_LABELS: Record<PropertyType, string> = {
  appartement: 'Appartement',
  maison: 'Maison',
  local_commercial: 'Local commercial',
  autre: 'Autre',
}

export const TYPOLOGY_OPTIONS = ['T1', 'T2', 'T3', 'T4', 'T5', 'T6', 'T7', 'T8', 'T9', 'T10'] as const

// Modes de chauffage — couvre les énergies courantes du parc français.
// (valeurs ≤ 30 car. : contrainte colonne heating_type VARCHAR(30))
export const HEATING_OPTIONS: { value: string; label: string }[] = [
  { value: 'individuel_gaz', label: 'Individuel : gaz' },
  { value: 'individuel_elec', label: 'Individuel : électrique' },
  { value: 'pompe_chaleur', label: 'Pompe à chaleur' },
  { value: 'fioul', label: 'Fioul' },
  { value: 'bois_granules', label: 'Bois / granulés (pellets)' },
  { value: 'collectif', label: 'Collectif (immeuble)' },
  { value: 'reseau_urbain', label: 'Réseau de chaleur urbain' },
  { value: 'autre', label: 'Autre' },
]

// Classes DPE A→G + « NS » (non soumis / vierge). Valeurs ≤ 2 car. (colonne VARCHAR(2)).
export const ENERGY_CLASSES: { value: string; label: string }[] = [
  { value: 'A', label: 'A : très performant' },
  { value: 'B', label: 'B' },
  { value: 'C', label: 'C' },
  { value: 'D', label: 'D' },
  { value: 'E', label: 'E' },
  { value: 'F', label: 'F' },
  { value: 'G', label: 'G : passoire thermique' },
  { value: 'NS', label: 'Non soumis / vierge' },
]

export type AmenityKey =
  | 'furnished' | 'kitchen_equipped' | 'has_elevator' | 'has_balcony'
  | 'has_terrace' | 'has_garden' | 'has_parking' | 'has_cellar' | 'has_fiber'
  | 'has_air_conditioning'

// Équipements / extérieurs — clé technique → libellé affiché
export const AMENITIES: { key: AmenityKey; label: string }[] = [
  { key: 'furnished', label: 'Meublé' },
  { key: 'kitchen_equipped', label: 'Cuisine équipée' },
  { key: 'has_fiber', label: 'Fibre internet' },
  { key: 'has_air_conditioning', label: 'Climatisation' },
  { key: 'has_elevator', label: 'Ascenseur' },
  { key: 'has_balcony', label: 'Balcon' },
  { key: 'has_terrace', label: 'Terrasse' },
  { key: 'has_garden', label: 'Jardin' },
  { key: 'has_parking', label: 'Parking / garage' },
  { key: 'has_cellar', label: 'Cave' },
]

export interface Property {
  id: string
  name: string
  reference: string | null
  address: string
  address2: string | null
  zip_code: string
  city: string
  country: string
  property_type: PropertyType
  full_address: string
  owner_id: string | null
  owner_user_id: string | null
  owner_name: string | null
  account_name?: string | null
  owner_national_id?: string | null
  owner_email: string | null
  owner_phone: string | null
  description: string | null
  notes: string | null
  year_built: number | null
  // ── Acquisition (achat ou construction) ────────────────────────────────────
  acquisition_date: string | null
  acquisition_value: number | null
  // ── Caractéristiques du bien ───────────────────────────────────────────────
  typology: string | null          // T1 … T10
  floor: number | null
  area_sqm: number | null
  bathrooms: number | null         // salles d'eau / de bain
  heating_type: string | null
  energy_class: string | null
  // ── Équipements & extérieurs ───────────────────────────────────────────────
  furnished: boolean
  kitchen_equipped: boolean
  has_elevator: boolean
  has_balcony: boolean
  has_terrace: boolean
  has_garden: boolean
  has_parking: boolean
  has_cellar: boolean
  has_fiber: boolean
  has_air_conditioning: boolean
  is_occupied: boolean
  is_available: boolean
  unit_count: number
  occupied_count: number
  created_at: string
  updated_at: string
}

export interface PropertyListItem {
  id: string
  name: string
  city: string
  property_type: PropertyType
  full_address: string
  owner_id: string | null
  owner_user_id: string | null
  owner_name: string | null
  account_name?: string | null
  typology: string | null
  area_sqm: number | null
  is_occupied: boolean
  unit_count: number
  occupied_count: number
  created_at: string
}

File: backend/app/schemas/property.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, EmailStr

from app.models.property import PropertyType


class PropertyCreate(BaseModel):
    name: str
    reference: str | None = None
    address: str
    address2: str | None = None
    zip_code: str
    city: str
    country: str = "France"
    property_type: PropertyType = PropertyType.APPARTEMENT
    owner_id: uuid.UUID | None = None
    owner_user_id: uuid.UUID | None = None
    owner_name: str | None = None
    owner_email: EmailStr | None = None
    owner_phone: str | None = None
    description: str | None = None
    notes: str | None = None
    year_built: int | None = None
    acquisition_date: date | None = None
    acquisition_value: float | None = None
    # ── Caractéristiques du logement ──────────────────────────────────────────
    typology: str | None = None  # T1 … T10
    floor: int | None = None
    area_sqm: float | None = None
    bathrooms: int | None = None  # salles d'eau / de bain
    heating_type: str | None = None
    energy_class: str | None = None
    # ── Équipements & extérieurs ──────────────────────────────────────────────
    furnished: bool = False
    kitchen_equipped: bool = False
    has_elevator: bool = False
    has_balcony: bool = False
    has_terrace: bool = False
    has_garden: bool = False
    has_parking: bool = False
    has_cellar: bool = False
    has_fiber: bool = False
    has_air_conditioning: bool = False


class PropertyUpdate(BaseModel):
    name: str | None = None
    reference: str | None = None
    address: str | None = None
    address2: str | None = None
    zip_code: str | None = None
    city: str | None = None
    country: str | None = None
    property_type: PropertyType | None = None
    owner_id: uuid.UUID | None = None
    owner_user_id: uuid.UUID | None = None
    owner_name: str | None = None
    owner_email: EmailStr | None = None
    owner_phone: str | None = None
    description: str | None = None
    notes: str | None = None
    year_built: int | None = None
    acquisition_date: date | None = None
    acquisition_value: float | None = None
    typology: str | None = None
    floor: int | None = None
    area_sqm: float | None = None
    bathrooms: int | None = None
    heating_type: str | None = None
    energy_class: str | None = None
    furnished: bool | None = None
    kitchen_equipped: bool | None = None
    has_elevator: bool | None = None
    has_balcony: bool | None = None
    has_terrace: bool | None = None
    has_garden: bool | None = None
    has_parking: bool | None = None
    has_cellar: bool | None = None
    has_fiber: bool | None = None
    has_air_conditioning: bool | None = None


class PropertyResponse(BaseModel):
    id: uuid.UUID
    ref_code: str | None = None
    name: str
    reference: str | None
    address: str
    address2: str | None
    zip_code: str
    city: str
    country: str
    property_type: PropertyType
    full_address: str
    owner_id: uuid.UUID | None = None
    owner_user_id: uuid.UUID | None = None
    owner_name: str | None
    owner_email: str | None
    owner_phone: str | None
    description: str | None
    notes: str | None
    year_built: int | None
    acquisition_date: date | None = None
    acquisition_value: float | None = None
    typology: str | None = None
    floor: int | None = None
    area_sqm: float | None = None
    bathrooms: int | None = None
    heating_type: str | None = None
    energy_class: str | None = None
    furnished: bool = False
    kitchen_equipped: bool = False
    has_elevator: bool = False
    has_balcony: bool = False
    has_terrace: bool = False
    has_garden: bool = False
    has_parking: bool = False
    has_cellar: bool = False
    has_fiber: bool = False
    has_air_conditioning: bool = False
    is_occupied: bool = False
    is_available: bool = True
    unit_count: int = 0
    occupied_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PropertyListItem(BaseModel):
    id: uuid.UUID
    ref_code: str | None = None
    name: str
    city: str
    property_type: PropertyType
    full_address: str
    owner_id: uuid.UUID | None = None
    owner_user_id: uuid.UUID | None = None
    owner_name: str | None
    typology: str | None = None
    area_sqm: float | None = None
    is_occupied: bool = False
    unit_count: int = 0
    occupied_count: int = 0
    created_at: datetime

    model_config = {"from_attributes": True}
